library/swift/RequestMethod.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// Represents an HTTP request method.
+@objc
+public enum RequestMethod: Int {
+  case delete
+  case get
+  case head
+  case options
+  case patch
+  case post
+  case put
+  case trace
+
+  /// String representation of this method.
+  var stringValue: String {
+    switch self {
+    case .delete:
+      return "DELETE"
+    case .get:
+      return "GET"
+    case .head:
+      return "HEAD"
+    case .options:
+      return "OPTIONS"
+    case .patch:
+      return "PATCH"
+    case .post:
+      return "POST"
+    case .put:
+      return "PUT"
+    case .trace:
+      return "TRACE"
+    }
```

Comment:
super nit, but if we put these all in a const array indexed by the enum value, we can avoid all the branching generated by the switch.

Reply:
We definitely could do that, my only concern is that it's a bit fragile for cases where we add a new log level (if someone were to forget to add it to the string array). Do you think it's worth trading compiler-enforcement that each type has a `String` value in an exhaustive `switch`?

The alternative would be something like this:

```swift
/// List of log levels indexed by `LogLevel.rawValue`.
private let kLogLevelStringValues = [
  "trace",
  "debug",
  "info",
  "warn",
  "error",
  "critical",
  "off",
]

/// Available logging levels for an Envoy instance.
/// Note that some levels may be compiled out.
@objc
public enum LogLevel: Int {
  case trace = 0
  case debug
  case info
  case warn
  case error
  case critical
  case off

  /// String representation of the log level.
  var stringValue: String {
    return kLogLevelStringValues[self.rawValue]
  }
}
```

Reply:
I mean, it's definitely a silly micro-optimization. Since I think this will very rarely change, my slight preference is still to do the optimization, but I think you could make the very fair case that the compiler check on the switch statement is more valuable. I'm good with however you'd like to proceed.

Reply:
I'd prefer to leave it like this if you don't feel strongly. Since this codepath is rarely traversed (once per startup of Envoy at the moment), I think using a `switch` instead of a potentially faster array lookup is a good tradeoff for compile-time safety